code/controllers/subsystem/security_level.dm

```diff
@@ -56,6 +56,13 @@ SUBSYSTEM_DEF(security_level)
 		SSshuttle.emergency.modTimer(old_shuttle_call_time_mod)
 		SSshuttle.emergency.modTimer(selected_level.shuttle_call_time_mod)
 
+	//SKYRAT EDIT ADDITION START- In the event of an orange alert, change the airlock permissions
+	if (istype(current_security_level, /datum/security_level/orange))
```

Comment:
i'm gonna just re-apply the comment if you're gonna hide it

this is not what I meant when I said to make it use signals

please make doors that you want to be emergency opened by engineers register a proc to the signal of the alert level changing, and then have them adjust their status based on that

if you need help doing this, reach out in the coderbus discord or at the /tg/ development channels and I can walk you through what you would need to do to listen to the signal to make the changes

Reply:
On two attempts registering all the airlocks to the signal kills the server, but I'll try again.

Reply:
signal code should not be this poorly performant, if it is we have a bug to resolve upstream, if it occurs again push the code and send me a link and I'll step through it

Reply:
Moved it to the check signal proc